### PR TITLE
Generic propulsion

### PR DESCRIPTION
**Pull request by christophe-david**
The objective of this PR is to make the mission performance model as independent as possible of propulsion model.

This is done by delegating to the propulsion model the task of computing the mass consumption.

The developer of a new propulsion model will have the possibility to add any parameter to the computed flight points and his implementation of the mass consumption may use them (or not).
The documentation has been updated to explain how to do that (see https://fast-oad.readthedocs.io/en/generic-propulsion/documentation/custom_modules/add_propulsion_module.html).

### CODE REVIEW

File: docs/documentation/custom_modules/add_propulsion_module.rst [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    >>> flight_points = [ FlightPoint(row) for row in df.iloc ]  # a list of FlightPoint instances from the whole DataFrame
+
+
+.. _flight_point_extensibility:
+
+Extensibility
+=============
+If you need FlightPoint to store parameters that are not already there, you
+can do this in your Python code (outside of any class or function, preferably
+in the python module where you will use these new paramaters)::
```
Comment (ScottDelbecq) [inline docs/documentation/custom_modules/add_propulsion_module.rst:64]: typo: paramaters

File: src/fastoad/models/performances/mission/segments/base.py [hunk 2]
```diff
@@ -210,7 +210,7 @@ def compute_next_flight_point(
         previous = flight_points[-1]
         next_point = FlightPoint()
 
-        next_point.mass = previous.mass - previous.sfc * previous.thrust * time_step
+        next_point.mass = previous.mass - self.propulsion.get_consumed_mass(previous, time_step)
```
Comment (ScottDelbecq) [inline src/fastoad/models/performances/mission/segments/base.py:213]: La je vois vraiment que tu l'as rendu générique ! C'est parfait.

**Review by ScottDelbecq — APPROVED** [review]
RAS
Comme ça je vois bien la généricité et la possibilité de basculer sur de la propu élec.
On vera en testant !

File: docs/documentation/custom_modules/add_propulsion_module.rst (hunk 1 above)
Comment (christophe-david) [reply docs/documentation/custom_modules/add_propulsion_module.rst]: Merci. Corrigé en 726314aff0e0e05bb9ff2d2c4f87ed0d18fbbc4f

**Pull request merged**